homeassistant/components/shelly/entity.py

```diff
@@ -263,12 +267,67 @@ async def set_state(self, **kwargs: Any) -> Any:
             return None
 
 
+class ShellyRpcEntity(entity.Entity):
+    """Helper class to represent a rpc entity."""
+
+    def __init__(self, wrapper: RpcDeviceWrapper, key: str) -> None:
+        """Initialize Shelly entity."""
+        self.wrapper = wrapper
+        self.key = key
+        self._attr_should_poll = False
+        self._attr_device_info = {
+            "connections": {(device_registry.CONNECTION_NETWORK_MAC, wrapper.mac)}
+        }
+        self._attr_unique_id = f"{wrapper.mac}-{key}"
+        self._attr_name = get_rpc_entity_name(wrapper.device, key)
+
+    @property
+    def available(self) -> bool:
+        """Available."""
+        return self.wrapper.device.connected
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to HASS."""
+        self.async_on_remove(self.wrapper.async_add_listener(self._update_callback))
+
+    async def async_update(self) -> None:
+        """Update entity with latest info."""
+        await self.wrapper.async_request_refresh()
+
+    @callback
+    def _update_callback(self) -> None:
+        """Handle device update."""
+        self.async_write_ha_state()
+
+    async def set_state(self, method: str, params: Any) -> Any:
+        """Set block state (RPC request)."""
```

Comment:
This comment is confusing.

Reply:
changed to `call_rpc` and updated comment